### ISSUE TITLE
[standalone] vPBX can not do external/internal calls

### ISSUE THREAD

**Issue by gmoreno90** [irontec/ivozprovider#2728]
### Description
I'm trying to create a vPbx Client and this client cal recieve call but not sent (internal or external)
We configured:

- 2 Users (with the correct setup of extension and terminal)
- 2 Terminals (both registered OK)
- 2 Extensions (1001 and 1002)
- 1 DID (inbound and outbond)
- 1 Hung Groups (We use to recieve calls and thats works)

When try to call between 1001 to 1002 return 603 Decline
searching the asterisk debug we can see the following error
``
       > [2dec43c3] Executing [1002@trunks:1] NoOp("PJSIP/proxytrunks-0000001c", "Incoming external call from "" <gonzalo> to 1002") in new stack
       > [2dec43c3] Executing [1002@trunks:2] AGI("PJSIP/proxytrunks-0000001c", "agi://127.0.0.1:4573/fastagi-runner.php?command=Dialplan/Trunks") in new stack
       > [2dec43c3] [Kernel.php:60            ] AGI Error ("PJSIP/proxytrunks-0000001c", "DDI 1002 not found in database.")
    -- <PJSIP/proxytrunks-0000001c>AGI Script agi://127.0.0.1:4573/fastagi-runner.php?command=Dialplan/Trunks completed, returning 0
    -- Auto fallthrough, channel 'PJSIP/proxytrunks-0000001c' status is 'UNKNOWN'
``

How we can configure a vpbx on that version? or is samething missing on configuration side?

Thanks

---

- IvozProvider version: 4.0~4.0.5~20240430.1111~34aa327+tempest

- Related sections: 
    - [ ] SIP & Telephony
    - [X] vPBX Functionalities
    - [ ] Billing & Invoicing
    - [ ] Web Interface
    - [ ] API
    - [ ] Other

### Troubleshooting
Create the following data on vpbx client
- 2 Users (with the correct setup of extension and terminal)
- 2 Terminals (both registered OK)
- 2 Extensions (1001 and 1002)
- 1 DID (inbound and outbond)
- 1 Hung Groups (We use to recieve calls and thats works)

### Additional Information
N/A

**Comment by gmoreno90** [comment]
Hello!
I make the same test using the stable version and have the same result.

![image](https://github.com/user-attachments/assets/d1a87da8-459a-47c8-9f57-f683e65e45ed)

IvozProvider version: 3.4~3.4.1~20240212.929~ee6564e+halliday

That test was made with the Demo Brand and vpbx demo client

**Comment by gmoreno90** [comment]
Hello!
I make the same on 4.1 version and have the same result.

![image](https://github.com/user-attachments/assets/721f5bea-ae3d-4b7f-ad92-0cf7a11ad726)

IvozProvider version: 4.1~4.1.0~20240805.1214~83ffdec+tempest
That test was made with the Demo Brand and vpbx demo client

**Comment by Kaian** [comment]
Hi @gmoreno90 

Are you using different IP address for Trunks and Users kamailio proxies?
It looks like asterisk is identifing your user calls as trunks (external) calls 

Also, I would also try updating to 4.2.0 version (It will be released today) that have some fixed on how proxies are identified by asterisk on standalone installations.

Best regards

**Comment by gmoreno90** [comment]
Hi @Kaian 
I'm using the same IP for both proxies.
I intall a the new version, and now the proxytrunks not start with the following error. need to open a new ticket? with that issue?

root@ivoz42:/etc/apt/sources.list.d# systemctl -l status kamailio@trunks.service --no-pager
× kamailio@trunks.service - Kamailio - the Open Source SIP Server
     Loaded: loaded (/lib/systemd/system/kamailio@.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Thu 2024-09-26 14:49:16 CEST; 6min ago
        CPU: 137ms

Sep 26 14:49:16 ivoz42 kamailio[25997]:              tcp: trunks.ivozprovider.local [46.183.118.130]:8002
Sep 26 14:49:16 ivoz42 kamailio[25997]:              tcp: trunks.ivozprovider.local [46.183.118.130]:8001
Sep 26 14:49:16 ivoz42 kamailio[25997]:              tls: trunks.ivozprovider.local [46.183.118.130]:5061
Sep 26 14:49:16 ivoz42 kamailio[25997]: Aliases:
Sep 26 14:49:16 ivoz42 /usr/sbin/kamailio[25999]: WARNING: db_mysql [km_my_con.c:187]: db_mysql_new_connection(): opt_ssl_ca option not supported by mysql version (value (null)) - ignoring
Sep 26 14:49:16 ivoz42 /usr/sbin/kamailio[25999]: ERROR: dmq [dmq.c:241]: mod_init(): server_uri is not a socket the proxy is listening on
Sep 26 14:49:16 ivoz42 /usr/sbin/kamailio[25999]: ERROR: <core> [core/sr_module.c:1039]: init_mod(): Error while initializing module dmq (/usr/lib/x86_64-linux-gnu/kamailio/modules/dmq.so)
Sep 26 14:49:16 ivoz42 systemd[1]: kamailio@trunks.service: Control process exited, code=exited, status=255/EXCEPTION
Sep 26 14:49:16 ivoz42 systemd[1]: kamailio@trunks.service: Failed with result 'exit-code'.
Sep 26 14:49:16 ivoz42 systemd[1]: Failed to start kamailio@trunks.service - Kamailio - the Open Source SIP Server.

Best Regards

**Comment by Kaian** [comment]
Hi,

Can you provide the contents of /etc/kamailio/proxytrunks/listeners.cfg and /etc/kamailio/proxytrunks/ports.cfg ?

Best Regards,

**Comment by Kaian** [comment]
Hi,

Could you check if this can be reproduced in 4.2.0?

Best regards,

**Comment by insxa** [comment]
Mine is:
`# proxytrunks (id: 1)
listen=udp:trunks:SIP_PORT
listen=tcp:trunks:SIP_PORT
listen=tls:trunks:SIPS_PORT
listen=tcp:trunks:XMLRPC_PORT
listen=tcp:trunks:RPC_PORT`

`#!define TRUNKS_DMQ_SERVER "sip:trunks.ivozprovider.local:5060"
#!define USERS_DMQ_SERVER "sip:users.ivozprovider.local:5060"

#!define SIP_PORT 5060
#!define SIPS_PORT 5061
#!define RPC_PORT 8001
#!define XMLRPC_PORT 8002
`

**Comment by insxa** [comment]
I have latest installed, with 2 public IPs, set sip domain and registered both sips to different linphone android and desktop apps. user alice and bob. cant call extensions 101/102 i get 407 / Proxy Authentification Required

**Comment by insxa** [comment]
How to check realm and passwords?

**Comment by gmoreno90** [comment]
Hi @Kaian I tested again with v 4.2.0 and the result is the same.

the cat of the files are the following

root@testivoz:~# cat /etc/kamailio/proxytrunks/listeners.cfg
# proxytrunks (id: 1)
listen=udp:trunks:SIP_PORT
listen=tcp:trunks:SIP_PORT
listen=tls:trunks:SIPS_PORT
listen=tcp:trunks:XMLRPC_PORT
listen=tcp:trunks:RPC_PORT

root@testivoz:~# cat /etc/kamailio/proxytrunks/ports.cfg
#!define TRUNKS_DMQ_SERVER "sip:trunks.ivozprovider.local:5060"
#!define USERS_DMQ_SERVER "sip:users.ivozprovider.local:5060"

#!define SIP_PORT 5060
#!define SIPS_PORT 5061
#!define RPC_PORT 8001
#!define XMLRPC_PORT 8002

maybe need to do a manual adjust?

**Comment by gmoreno90** [comment]
hi @Kaian could you help with that?
Many thanks

**Comment by Kaian** [comment]
Hi,

No manual configuration should be required. There is a script (/etc/kamailio/autoconf) specified in the systemd unit file that runs before starting Kamailio. This script updates all configuration files with the information from the database configured through the platform portal.

Additionally, ensure the local DNS server is working and that you can ping both users.ivozprovider.local and trunks.ivozprovider.local

Regards

**Comment by gmoreno90** [comment]
Hi @Kaian I tested with the last version and the error is the same,
The local DNS server is working and can ping to both endpoints.
If I tested the version using 2 different ips, works fine, but with 1 Ip fail

Thanks for review

**Comment by Kaian** [comment]
Hi!

Can you provider a PCAP file and Asterisk dialplan log for a Alice to Bob call?

User calls should have a X-Info-Endpoint header to proprely identify the dialog as user call no matter message source IP address and the dialplan log should be different that the one provided in the screenshots above.

Since #2756 user endpoints identifies should look like this:
```
*CLI> pjsip show identify b1c1t1_alice 

 Identify:  <Identify/Endpoint...........................................................>
      Match:  <criteria...........................>
==========================================================================================

 Identify:  b1c1t1_alice/b1c1t1_alice
     Header: X-Info-Endpoint: b1c1t1_alice


 ParameterName     : ParameterValue
 =================================================
 endpoint          : b1c1t1_alice
 match             : 
 match_header      : X-Info-Endpoint: b1c1t1_alice
 match_request_uri : 
 srv_lookups       : false

````


Best Regards,

**Comment by gmoreno90** [comment]
I will close becouse in new versions works. Thanks
